### DISCUSSION TITLE
Logging in c:geo

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#10383] [Q&A] [answered]
A question which might be unqualified:
In logfiles over the last months I keep seeing tons of this
```
I/cgeo.geocachin: Explicit concurrent copying GC freed 4369(461KB) AllocSpace objects, 0(0B) LOS objects, 66% free, 16MB/48MB, paused 95us total 81.850ms```
```

Is this a helpful information log or some leftover of a debugging?
IMHO it pollutes the log, so if its not really needed it could/should be removed or (if needed after all) limited to debug?!

Another one would be, that most logs I receive on support start with warnings about Locus not being installed logging the whole stacktracke. IMHO this is also kind of spam. Couldn't this be logged just as a one line warning and maybe even only in debug mdoe?

To make this question even a bigger one:
Do we have a strategy or is there any Android recommendation about logging levels?
From the past I was always very happy, that the debug log allowed me to see whats happening under the hood by e.g. also providing info about successful tasks and network communication. 
But I somehow have the impression that some newer functions are rather silent and output no debug messages. 

Sorry for those questions, but I have no deep idea about how these logging and log levels work.

**Comment by bekuno** [comment]
> Another one would be, that most logs I receive on support start with warnings about Locus not being installed logging the whole stacktracke. IMHO this is also kind of spam. Couldn't this be logged just as a one line warning and maybe even only in debug mdoe?

I created issue #10385 for that problem.
This has to be fixed in the Locus connector.

**Comment by eddiemuc** [comment, marked as answer]
Wrt to the "Explicit concurrent copying GC" message: this log statement is issued whenever Garbage collection is triggered. The "explicit" in the message suggests that gc was somehow triggered explicitely by application code (c:geo or one of its used libraries).

I found this page explaining how to interpret the messages: https://blog.gceasy.io/2017/05/09/understanding-android-gc-logs/ 

I don't know however what code part exactly triggers this more often than before and why.
